Remove unused code

panslava opened the pull request:
/assign @aojea

panslava commented:
/test cloud-provider-gcp-verify-up-to-date

aojea commented:
it fails to compile now, hence the jobs are failing, you may need to run `gazelle` or some of the update scripts

panslava commented:
oh I accidentally commited next work, that was not supposed to go in this commit, fixing

bowei commented:
was this done using static analysis?

panslava commented:
@bowei yes, I used golangci-lint "unused" linter, I can add it to the repo

aojea commented on cmd/gcp-controller-manager/main.go:
```diff
@@ -72,7 +72,6 @@ var (
 	authSyncNodeURL                       = pflag.String("auth-sync-node-url", "", "URL for reaching the Auth Service SyncNode API.")
 	hmsAuthorizeSAMappingURL              = pflag.String("hms-authorize-sa-mapping-url", "", "URL for reaching the Hosted Master Service AuthorizeSAMapping API.")
 	hmsSyncNodeURL                        = pflag.String("hms-sync-node-url", "", "URL for reaching the Hosted Master Service SyncNode API.")
-	autopilotEnabled                      = pflag.Bool("autopilot", false, "Is this a GKE Autopilot cluster.")
```
if you remove flags, independently they maybe a noop, the manifests using this binary need to be updated or will fail to execute

aojea commented:
I'm not much of a fun of removing code that is unused just because of that, it is not we are charged by number of lines and there is always something that may go wrong ... do we really need to take this risk?

panslava commented:
yeh, tbh, initially I thought that would help me for further stuff, but not really

I remade this pr and ran linter with slightly softer (or stricter lol) rules so now I removed code that should not break anything. Also added linter config

but wouldn't say it's really needed, so feel free to ignore and close

bowei commented:
We should try to delete dead code (otherwise when we will do that?) but flags we need to mark as deprecated or whatnot and keep for a while.

panslava commented:
okay then

1. brought back removing all the code that seems to be unused
2. Deprecated the autopilot flag, but didn't remove

aojea commented on .golangci.yaml:
```diff
@@ -0,0 +1,11 @@
+linters:
+  # Disable all linters.
+  # Default: false
+  disable-all: true
```
do we lose coverage?

we were running the linters before AFAIK as part of https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/cloud-provider-gcp/755/cloud-provider-gcp-verify-all/1828578606167625728
